packages/venia-concept/src/actions/cart/asyncActions.js

```diff
@@ -125,6 +125,64 @@ export const addItemToCart = (payload = {}) => {
     };
 };
 
+export const removeItemFromCart = payload => {
+    const { item } = payload;
+
+    return async function thunk(dispatch, getState) {
+        dispatch(actions.removeItem.request(payload));
+
+        try {
+            const { cart } = getState();
+            const { guestCartId } = cart;
+            const cartItemCount = cart.details ? cart.details.items_count : 0;
+
+            if (!guestCartId) {
+                const missingGuestCartError = new Error(
+                    'Missing required information: guestCartId'
+                );
+                missingGuestCartError.noGuestCartId = true;
+                throw missingGuestCartError;
+            }
+
+            const cartItem = await request(
+                `/rest/V1/guest-carts/${guestCartId}/items/${item.item_id}`,
+                {
+                    method: 'DELETE'
+                }
+            );
+            // When removing the last item in the cart, perform a reset
+            // to prevent a bug where the next item added to the cart has
+            // a price of 0
```

Comment:
Good comment; this would otherwise be really unintuitive. That's got to be a server-side bug.